app/javascript/Pages/Connections.jsx

```diff
@@ -1,40 +1,104 @@
 import React from "react";
+import axios from "axios";
 import PropTypes from "prop-types";
-import { Head, Link } from "@inertiajs/inertia-react";
+import { Inertia } from "@inertiajs/inertia";
 import Layout from "./Layout";
+import { useState } from "react";
+import { FaTimes } from "react-icons/fa";
 
 export default function Welcome(props) {
-  const ConnectionsList = (props) => {
-    const connectionsListChild = props.connections.map((c) => {
-      if (c.match.student) {
-        return <li>Student: {c.match.student.username}</li>;
-      } else {
-        return <li>Teacher: {c.match.skill_class.teacher.username}</li>;
-      }
-    });
+ const avatar = (image) => {
+  return `${image}?tr=w-60,h-60,r-max`;
+ };
+
+ const endConnectionHandler = (connection_id) => {
+  axios
+   .put(`/api/connections/${connection_id}/`, {
+    status: "given",
+    headers: window.defaultHeaders,
+   })
+   .then(() => {
+    Inertia.get("/connections");
+   })
+   .catch((error) => {
+    if (error.response.data.error) {
+     alert(error.response.data.error);
+    }
+   });
+ };
 
-    return <ul>{connectionsListChild}</ul>;
-  };
+ const otherUser = (connection) => {
+  if (
+   props.current_user.id == connection.match.student.id
+  ) {
+   //Current user is student, return teacher
+   return connection.match.skill_class.teacher;
+  } else if (
+   props.current_user.id ==
+   connection.match.skill_class.teacher.id
+  ) {
+   //Current user is teacher, return student
+   return connection.match.student;
+  }
+ };
 
-  return (
-    <Layout current_user={props.current_user}>
-      <Head title="Welcome" />
-      <h1>Connections</h1>
-      <p>Student connections</p>
-      <ConnectionsList connections={props.student} />
-      <p>Teacher connections</p>
-      <ConnectionsList connections={props.teacher} />
+ const [selectedConnection, setSelectedConnection] =
+  useState({});
 
-      <Link href="/home" headers={window.defaultHeaders} as="button">
-        Home
-      </Link>
-    </Layout>
-  );
+ return (
+  <Layout current_user={props.current_user}>
+   <h1>Connections</h1>
+   {props.connections.map((c, i) => {
+    return (
+     <div key={i}>
+      <img src={avatar(otherUser(c).avatar.url)} />
+      <p>{otherUser(c).name}</p>
+      <p>{otherUser(c).username}</p>
+      <button
+       onClick={() =>
+        setSelectedConnection({
+         id: c.id,
+         name: otherUser(c).name,
+        })
+       }
+      >
+       Bye!
+      </button>
+     </div>
+    );
+   })}
+
+   {selectedConnection.id ? (
```

Comment:
I think you can do `selectedConnection.id && ...` here

Reply:
Instead of `selectedConnection.id ? (...) : null` you can do `selectedConnection.id && (...)` and avoid the `: null`